client/src/hooks/useDownloadOptionsContext.js

```diff
@@ -46,7 +48,18 @@ export const useDownloadOptionsContext = () => {
 
   // Get the first computed file that matches modality and format
   const getFoundFile = (files = computedFiles) =>
-    files.find((file) => file.modality === modality && file.format === format)
+    files.find(
+      (file) =>
+        file.modality === modality &&
+        file.format === format &&
+        file.includes_merged === selectedMerged
+    )
+  // Check if the user-selected modality is in 'mergedObjectsKeys'
+  const isMergedObjectKey = mergedObjectsKeys.includes(modality)
+
+  // Check the availability of the merged objects based on modalities and 'includes_merged' flag per computed file
+  const isMergedObjectsAvailable =
+    isMergedObjectKey && computedFiles.some((cf) => cf.includes_merged)
```

Comment:
Please remove `isMergedObjectKey` this will still always be correct with the `computedFiles.some` check

Reply:
This check is necessary to make sure that the merged objects checkbox is disabled for unsupported modalities (i.e., `SPATIAL`). 

Otherwise, for the project containing the spatial data, the merged objects checkbox will be selectable for both `SINGLE_CELL` and `SPATIAL` modalities (pls see [`SCPCP000006`](https://api.staging.scpca.alexslemonade.org/v1/projects/SCPCP000006)) , since `computedFiles.some` check will be true.

Please let me know!

Reply:
As chatted on Eng scrum, I've applied feedback and no longer use `modality` for  checking the merged objects availability on FE. Pls see the update at e13e69edd65758c0aa6dd8ad1fb00a9aafb16d98, thank you!